### ISSUE TITLE
Hardcoded amd64 packages in tftp class

### ISSUE THREAD

**Issue by GregSutcliffe** [theforeman/puppet-foreman_proxy#346]
I've just been playing with setting up a RaspberryPi based Foreman appliance, and hit a minor snag. In the TFTP class we hardcode the amd64 versions of the EFI Grub packages - this [doesn't exist](https://packages.debian.org/jessie/grub-efi-amd64-bin) for ARM boards. There is an [equivalent ARM package](https://packages.debian.org/jessie/grub-efi-arm-bin) though.

What's the best solution here? Parameterise the EFI packages? Or maybe just add an $manage_efi flag and surround the whole EFI block with it?

**Comment by GregSutcliffe** [comment]
Opps, forgot the link to the tftp class - https://github.com/theforeman/puppet-foreman_proxy/blob/master/manifests/tftp.pp#L103

**Comment by mmoll** [comment]
AFAICT, the whole EFI implementation of all parts is amd64 only at the moment, but PRs welcome. :)

**Comment by ekohl** [comment]
https://github.com/theforeman/puppet-foreman_proxy/pull/347 does a bit in preparing for this by making the package names a variable.

**Comment by ekohl** [comment]
Right now I don't know which packages we should install on other arches

**Comment by GregSutcliffe** [comment]
https://packages.debian.org/search?suite=stable&section=all&arch=any&searchon=names&keywords=grub-efi is a list for debian at least. I think the issue is knowing which is right at any time. The architectire fact may not be reliable, eg, for picking between grub-efi-arm-bin and grub-efi-arm64-bin ...

**Comment by ekohl** [comment]
I like the ```foreman-bootloaders``` package because packaging is a much better place to solve architecture.

**Comment by GregSutcliffe** [comment]
Agreed, although then it has to exist for all the OSs we want to support. Which in my case is Arch ... ;)
